### ISSUE TITLE
Match Project CRS results in empty map

### ISSUE THREAD

**Issue by Llaves** [tomchadwin/qgis2web#860]
I am trying to export a map that contains two layers (one is a duplicate of the other, but with different symbology). If I leave the box "Match Project CRS" unchecked, the map exports with no problem. If I check the box, however, I get a blank map - no features and no layers legend. The project is in USA_Contiguous_Albers_Equal_Area_Conic, EPSG102003. The layers, however are in

    USER:100026 - * Generated CRS (+proj=aea +lat_1=29.5 +lat_2=45.5 +lat_0=37.5 +lon_0=-96 +x_0=0 +y_0=0 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs) - Projected.

I've attempted to export the layer (as shapefile) into EPSG102003, but when I reload it, it's still in the user CRS.

I suspect the layer CRS is at the root of the problem, but I can't figure out how to coerce it to save in the CRS I want.

Update:
I created a new project in which I imported the layer with the user crs and forced it to 102003 via Layer->Set CRS. I then set the project to match. Now both the layer and the project CRS are 102003. I clicked Match Project CRS and previewed. I get a blank canvas

QGIS 3.4.2, qgis2web 3.12.1, exporting as OpenLayers

**Comment by boesiii** [comment]
Can you share your data?  Why do you want to use a different CRS?

**Comment by Llaves** [comment]
Yes, I can share the data. Are you OK with a 3.5M shapefile, or would you prefer I try to find a reduced set that demonstrates the problem?
As to why a different CRS - my map is the entire US and in my qgis project I'm using the Conus Equal Area Albers projection. The default for qgis2web appears to be EPSG:3857, which is not area preserving. This creates substantial distortions of area.

**Comment by boesiii** [comment]
That size is fine.

**Comment by Llaves** [comment]
Here's the project file and the layer file. I think you should be able to unzip everything in place and have it work unless I somehow set paths to absolute. 
[MatchProjectCRS.zip](https://github.com/tomchadwin/qgis2web/files/4395717/MatchProjectCRS.zip)

**Comment by tomchadwin** [comment]
I don't know much about projections, butt he exported code doesn't identify one of your projections with a familiar EPSG code, but with an ESRI one:

```javascript
ol.proj.get("ESRI:102003").setExtent([-3182682.976803, -1732572.367161, 2778260.794236, 2506597.238520]);
```

Is it possible there is something slightly unusual about one of your projections, which OpenLayers plainly cannot support?

**Comment by Llaves** [comment]
While that specific projection might conceivably create a problem (though it's supported by Proj4), I tried a couple of other more common projections and have the same problem. I created a new map with source data in epsg:26913 (UTM 13N, NAD83). This is unambiguously a standard EPSG designation, the map and data source are both in that projection, yet I get an empty map when I export with Match Project CRS. I've attached that much simpler map here.

[ProjectCRS.zip](https://github.com/tomchadwin/qgis2web/files/4410467/ProjectCRS.zip)

**Comment by tomchadwin** [comment]
Hmm, you;re right. It's like it's missing the Proj4js definitions...

**Comment by Llaves** [comment]
Running the qgis2web output with debugger on generates an error on line 1 of layers.js - 
```
TypeError: ol.proj.get(...) is nulllayers.js:1:9
    <anonymous> file:///C:/Users/David/Desktop/ShortTerm/geo_temp/ProjectCRS/qgis2web_2020_03_31-11_26_24_270300/layers/layers.js:1
```

line 1 is the fetch of the projection - 
`ol.proj.get("EPSG:26913").setExtent([-38244.001992, 3447961.277734, 821295.399710, 4129877.981870]);`

Looking at this [sample code](https://openlayers.org/en/latest/doc/faq.html) from the OpenLayers website, I notice there is a register() function called after the proj4.defs() call.
```
proj4.defs('EPSG:21781',
  '+proj=somerc +lat_0=46.95240555555556 +lon_0=7.439583333333333 +k_0=1 ' +
  '+x_0=600000 +y_0=200000 +ellps=bessel ' +
  '+towgs84=660.077,13.551,369.344,2.484,1.783,2.939,5.66 +units=m +no_defs');
register(proj4);
```

**Comment by tomchadwin** [comment]
I think you are 100% correct - thanks so much for spotting and reporting the issue, and for finding the fix. To test, edit that output `layers.js`, and insert this as the very first line of the file:

```javascript
ol.proj.proj4.register(proj4);
```

I think that fixes it for me - can you test and confirm?

**Comment by tomchadwin** [comment]
Possible fix committed to master. Tested against both projects uploaded in this thread, and working. Please test and report back.

**Comment by tomchadwin** [comment]
Released as 3.13.0 and pushed to QGIS plugins repo.
